### ISSUE TITLE
Rule_L008 breaking query with unexpected output

### ISSUE THREAD

**Issue by PauloCarneiro99** [sqlfluff/sqlfluff#1120]
When I tried to apply sqlfluff fix on the query below I got warning messages and the sqlfluff produces an unexpected query as output.

## Input
```sql
select
    t.a ,t.b,sum(t2.c)
from table as t
join table2 as t2
    on t.a = t2.a
```

## Expected Behaviour
```sql
  SELECT
      t.a, t.b, SUM(t2.c)
  FROM table AS t
  JOIN table2 AS t2
      ON t.a = t2.a
```

## Observed Behaviour

```sql
SELECT
    t.a, t.b, SUMSUMSUMSUMSUMSUMSUMSUMSUMSUMsum(t2.c)
FROM table AS t
JOIN table2 AS t2
    ON t.a = t2.a
```

with the error: `WARNING  One fix for L008 not applied, it would re-cause the same error. `

## Steps to Reproduce

sqlfluff fix test.sql

## Version
Python 3.7.9
sqlfluff, version 0.5.6

## Configuration
```
[sqlfluff]
verbose = 0
dialect = ansi
rules=L002,L003,L004,L005,L006,L008,L010,L017,L018,L019,L030,L035,L042
sql_file_exts = .sql,.ddl

[sqlfluff:indentation]
indented_joins = False
template_blocks_indent = True

[sqlfluff:rules]
tab_space_size = 4
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper
```

**Comment by barrywhart** [comment]
Interesting! I've seen errors like this before when the code to detect the problem and/or place the fixes is slightly "off". My guess is that the rule may be assuming there's a space between the comma and `MAX`.

**Comment by scrambldchannel** [comment]
I had a bit of a look at this and noticed a couple of things:

* running `sqlfluff lint` shows _two_ linting errors for L008 line 2, pos 10 - should it not just report the issue once? 
* disabling L030 will prevent the bug from manifesting itself but I note that linting does not report an issue with L030

Is it somehow being caused by both problems (ie whitespace after the comma and capitalisation of `sum`) are at the same line/position in the file?

**Comment by barrywhart** [comment]
This is no longer happening in the latest `main` branch. I'm pretty sure this was the same issue as #1207, which was fixed in PR #1214.
